### DISCUSSION TITLE
Setup spectral in maven/gradle to build

### DISCUSSION THREAD

**Discussion by veer201989** [stoplightio/spectral#2249] [Q&A]
**User story.**
As a user, I can do lint my whole project while creating build using maven/gradle , so that I can identify the issues/warning/info.
**Is your feature request related to a problem?**
I am trying to lint my java project in which I have multiple YAML/JSON files, I want to lint all the files at the time of build but I am not able to get any setup process
**Describe the solution you'd like**
Please share the steps or process to setup spectral in maven/gradle build file.

**Additional context**

**Comment by veer201989** [comment]
Any one can have any kind of solution to build the OpenApi project using Gradle/maven build script?

**Reply by maboza12** [reply]
What is the gradle/maven plugin we would use for this?

**Reply by TSFenwick** [reply]
i found this for gradle https://github.com/michael-nestler/gradle-spectral I'm not sure how well supported it is

**Comment by nddipiazza-ebay** [comment]
i have created a Maven plugin for spectral, and will maintain it. It is now on maven central.
I will bump a new version for each major release of spectral.
https://github.com/nddipiazza/spectral-maven-plugin
-nicholas dipiazza
apache software foundation
